packages/metamask/src/index.ts

```diff
@@ -157,8 +157,9 @@ export class MetaMask extends Connector {
             method: 'wallet_switchEthereumChain',
             params: [{ chainId: desiredChainIdHex }],
           })
-          .catch((error: ProviderRpcError) => {
-            if (error.code === 4902 && typeof desiredChainIdOrChainParameters !== 'number') {
+          .catch((error: any) => {
```

Comment:
`ProviderRpcError` has `data` field on it, so you should be able to achieve your functionality w/o replacing it with `any`. If that becomes a problem for some other reason, we can always make that type a `generic`. But that shouldn't be required to get started.

Reply:
Thank your for the feedback! `data` is declared unknown types so I cannot dereference its properties. As you point  out there are a couple of ways to solve this. I added a different take that keeps the error type as `ProviderRpcError` Please let me know